### DISCUSSION TITLE
Remember me cookie required?

### DISCUSSION THREAD

**Discussion by nayakanirudh** [pocketbase/pocketbase#4356] [Q&A] [answered]
Sorry if my 2 questions are very basic. I could not find any related discussions in this forum

1. I am writing a simple webapp with alpine.js, FastAPI and Pocketbase. I am authenticating using client side scripting and interfacing directly with Pocketbase (exposed as a https website). Is this the intended way of using the Javascript SDK. Or am I doing this in an unsecure way. I love doing it this way because I can build things very quickly
2. Is a "Remember Me?" checkbox redundant with Pocketbase. Calling the auth api saves to local storage and I simply use the resultant PB token as a header in all of my routes

**Comment by ganigeorgiev** [comment, marked as answer]
> I am authenticating using client side scripting and interfacing directly with Pocketbase (exposed as a https website). Is this the intended way of using the Javascript SDK.

Yes, it is safe to interact directly with PocketBase as long as you have the proper collections API rules in place (this is also how I use it in my [other open source project](https://github.com/presentator/presentator)).

> Is a "Remember Me?" checkbox redundant with Pocketbase.

For most of the cases probably Yes, unless you want your users to be able to have temp/one-off "sesssions".
Also note that PocketBase is fully stateless and doesn't maintain session cookies or similar as other common backend services (we don't even store the tokens in the database).
A user is considered authenticated only if the request was sent with valid `Authorization:TOKEN` header (this header is added automatically by the SDK's if `pb.authStore.token` is set).
The "logout" action is just clearing the local authStore state (`pb.authStore.clear()`).

**Reply by nayakanirudh** [reply]
Amazing! Love the simplicity :)

**Reply by nayakanirudh** [reply]
If I can ask a follow-up: How do I make use of the PB token in other non-pocketbase parts of my application? I still need to check if the user is valid (let's say user token has expired after xx seconds) and I don't see an API to check token validity. I am currently calling the get users API and if the returned array is blank I return a 401

**Reply by ganigeorgiev** [reply]
We don't have a dedicated endpoint for checking the token state. For such cases I usually recommend sending an [auth refresh call](https://pocketbase.io/docs/api-records/#auth-refresh). On success it will return the latest user data and a new token with refreshed `exp` claim. It throws an error on invalid or expired token.
Depending on your API rules, the list users request could also work but if you don't need its result it could be slower than `getOne` or `authRefresh` calls (_both calls have almost identical performance_).
